### PR TITLE
[Feature] CX-3530 Announce image preview expander

### PR DESCRIPTION
**Pull request by josokinas**
# Problem
Currently on image preview screen reader does not announce when an image is being enlarged and enlargement closed.

# Solution
In order to announce a change of the "enlarged image" state, each toggle focuses on the container that has label toggled also between regular "alt" and "enlargedAlt" tags. ~~However, `aria-hidden` is used to force-read the label change instead of having a regular group announcement. In addition, to not have multiple of the same images on the DOM, regular image is hidden when "full screen" mode is on (that is, when preview is enlarged).~~

**EDIT by @stefaniacardenas:**
- Removed `aria-hidden` and added `role="img"` to inform the screen reader that the group is an image. This fixes the issue with the image label not being detected by screen readers.
- Added `aria-expanded=boolean` to inform the user that the image is expandable and its current state.

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [x] Have the CHANGELOG, README, MIGRATION, RELEASE_GUIDELINES docs been updated?
- [`n/a`] Have new automated tests been implemented?
- [`n/a`] Have new manual tests been written down?
- [x] Have tests passed locally?
- [x] Have any new strings been translated?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-653-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-653-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://653-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/EnlargedPreview/index.js [hunk 1]
```diff
@@ -47,23 +50,32 @@ class EnlargedPreview extends Component<Props, State> {
...
   )
 
   render() {
     const { isExpanded, hasEntered } = this.state
-    const { translate, src } = this.props
+    const { translate, src, altTag, enlargedAltTag } = this.props
     return (
       <div
+        ref={node => this.container = node}
+        tabIndex={-1}
```
[the remark this answers is not in the document]
Comment (stefaniacardenas) [reply src/components/EnlargedPreview/index.js]: We discussed the issue offline and it was fixed in [this commit](https://github.com/onfido/onfido-sdk-ui/pull/653/commits/f2506dfe71d0d7fc732b121dc666ce89861f17d8).

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-653-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-653-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://653-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/EnlargedPreview/index.js [hunk 2]
```diff
@@ -47,37 +50,49 @@ class EnlargedPreview extends Component<Props, State> {
...
+          aria-label={isExpanded ? enlargedAltTag : altTag}
+          aria-hidden="true" // hide regular group announcement in order to have label announced on each focus
+        >
+          {
+            isExpanded &&
+              <Pannable
+                ref={ node => this.image = node }
+                className={style.imageContainer}
+              >
+                <img onLoad={this.handleImageLoad} className={style.image} src={src} alt={enlargedAltTag} />
```
Comment (stefaniacardenas) [inline src/components/EnlargedPreview/index.js]: This alt tag is never read by the screenreader. It's only the parent aria-label that gets announced but there is no mention of the image when you are on the enlarged screen, which I think can be confusing to accessibility users.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://653-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-653-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-653-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://653-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Router/StepsRouter.js [hunk 3]
```diff
@@ -8,13 +8,8 @@ import { withFullScreenState } from '../FullScreen'
...
-    // Re-focus to container is needed for accessibility to have a correct
-    // tabbing order, and should be triggered when...
-    if (this.container &&
-        // ...step changes (for tabbing order to start from the top)
-        (prevProps.step !== this.props.step) ||
-        // ..."full screen" mode changes (e.g. for enlarged image preview)
-        (prevProps.isFullScreen !== this.props.isFullScreen)) {
+    // Re-focus to container when step changes in order for accessible tabbing
+    if (this.container && prevProps.step !== this.props.step) {
       this.container.focus()
```
Comment (rfreitas) [inline src/components/Router/StepsRouter.js]: so do we still want this focus to happen on the container?

Comment (stefaniacardenas) [reply src/components/Router/StepsRouter.js]: I will be handling this in another PR

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-653-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-653-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-653-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-653-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://653-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-653-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-653-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://653-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by seewah** [comment]
the enlarged image does not get read out on android/chrome/voiceover after clicking on Enlarge

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-653-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-653-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://653-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by seewah** [comment]
some code comment seems to be rendered in the enlarged view

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-653-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-653-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://653-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by seewah** [comment]
i am testing https://653-pr-onfido-sdk-ui-onfido.surge.sh on android and talkback and it seems the announcement only happens once. i.e. click enlarge -> [announcement] -> click close -> click enlarge -> [NO announcement]

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-653-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-653-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://653-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**